com.unity.render-pipelines.core/ShaderLibrary/UnityDOTSInstancing.hlsl

```diff
@@ -150,31 +150,43 @@ static const uint kDOTSInstancingFlagCameraMotion     = (1 << 2); // Object uses
 static const uint kDOTSInstancingFlagHasPrevPosition  = (1 << 3); // Object has a separate previous frame position vertex streams (e.g. for deformed objects)
 static const uint kDOTSInstancingFlagMainLightEnabled = (1 << 4); // Object should receive direct lighting from the main light (e.g. light not baked into lightmap)
 
+static DOTSVisibleData unity_SampledDOTSVisibleData;
+
+void SetupDOTSVisibleInstancingData()
+{
+    unity_SampledDOTSVisibleData = unity_DOTSVisibleInstances[unity_InstanceID];
```

Comment:
Have you checked the shader codegen produced by this approach compared to the previous version?

If the codegen is the same, then I think we can use this approach always. If the codegen is worse for some reason, then we could consider separating the regular `DOTS_INSTANCING_ON` variants from deferred material ones, and add some kind of macro here to use the static variable only in the latter case.

I would imagine the codegen of this approach should be good, but I think it would be a good idea to check since shader compilers can often be surprising. :)